source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc

```diff
@@ -142,23 +142,39 @@ void GrpcClientImpl::toAuthzResponseHeader(
   }
 }
 
-const Grpc::RawAsyncClientSharedPtr AsyncClientCache::getOrCreateAsyncClient(
+AsyncClientCacheSharedPtr AsyncClientCacheSingleton::getOrCreateAsyncClientCache(
+    Server::Configuration::FactoryContext& context,
```

Comment:
It is possible the context doesn't outlive the Singleton, so the returned AsyncClientCacheSharedPtr have dangling reference to them.

Reply:
Yeah, pass in the actual things that you need which are likely server scope and OK.

Reply:
Done